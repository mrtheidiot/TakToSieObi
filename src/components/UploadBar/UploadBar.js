import React, { useState } from "react";
import palceholder from "./../../assets/placeholder.png";
import storage from "../../firebaseConfig";
import { ref, uploadBytesResumable, getDownloadURL } from "firebase/storage";

import classes from "./UploadBar.module.css";

const UploadBar = (props) => {
  const [file, setFile] = useState("");
  const [percent, setPercent] = useState(0);
  const [fileLink, setFileLink] = useState(props.presetImage);



  function handleChange(event) {
    setFile(event.target.files[0]);
  }

  const handleUpload = () => {
    if (!file) {
      alert("Please upload an image first!");
    }

    const storageRef = ref(storage, `/files/${file.name}`);
    const uploadTask = uploadBytesResumable(storageRef, file);

    uploadTask.on(
      "state_changed",
      (snapshot) => {
        const percent = Math.round(
          (snapshot.bytesTransferred / snapshot.totalBytes) * 100
        );

        setPercent(percent);
      },
      (err) => console.log(err),
      () => {
        getDownloadURL(uploadTask.snapshot.ref).then((url) => {
          setFileLink(url);
          props.returnLink(url);
        });
      }
    );
  };

  return (
    <div className={classes.wrapper}>
      <section className={classes.image_container}>
        <img src={fileLink ? fileLink : palceholder} alt="uploaded image" />
      </section>
      <section className={classes.inputs_container}>
        <div className={classes.inputs}>
          <h3>Wybierz zdjęcie:</h3>
          <input
            type="file"
            onChange={handleChange}
            className={classes.input}
          />
          <button onClick={handleUpload}>Dodaj</button>
        </div>
        <div className={classes.loadingBar_container}>
          <div className={classes.loadingBar} style={{ width: `${percent}%` }}>
            .
          </div>
        </div>
      </section>
    </div>
  );
};

export default UploadBar;

// <img src={fileLink} />
// <input type="file" onChange={handleChange} className={classes.input} />
// <button className={classes.button} onClick={handleUpload}>
//   Dodaj
// </button>
// {fileLink && (
//   <button className={classes.button} onClick={handleRemove}>
//     Usuń
//   </button>
// )}
// <div className={classes.uploadBarOuter}>
//   <div className={barClasses} style={{ width: `${percent}%` }}>
//     ...
//   </div>
// </div>
