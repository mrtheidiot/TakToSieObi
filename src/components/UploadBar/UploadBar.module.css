.wrapper {
    width: 100%;
    border: solid 1px black;
    padding: 1%;
    display: flex;
    justify-content: space-between;
    height: 25px;
}

.input {
    width: 30%;
}

.button {
    width: 10%;
}

.uploadBarOuter {
    width: 50%;
}
.uploadBarInner {
    background-color: blue;
    color: white;
}

.changeColor {
    background-color: green;
}