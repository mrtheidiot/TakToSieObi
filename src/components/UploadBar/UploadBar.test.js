import { render } from "@testing-library/react";

import UploadBar from "./UploadBar";

it("matches the UploadBar snapshot", () => {
  const {
    container: { firstChild },
  } = render(<UploadBar />);
  
  expect(firstChild).toMatchSnapshot()
});
