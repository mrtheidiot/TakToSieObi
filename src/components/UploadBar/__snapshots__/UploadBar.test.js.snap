// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the UploadBar snapshot 1`] = `
<div
  class="wrapper"
>
  <div
    class="inputs"
  >
    <p>
      Wybierz zdjęcie:
    </p>
    <input
      class="input"
      type="file"
    />
    <button>
      Dodaj
    </button>
  </div>
  <div
    class="bar_container"
  >
    <div
      class="bar"
      style="width: 0%;"
    >
      <span
        style="display: none;"
      >
        Dodano pomyślnie!
      </span>
    </div>
  </div>
</div>
`;
